Multiple fixes for price-box:
Fix prices config data when there's no finalPrice form jsonConfig.
Skip tier prices calculations when tier prices are not available.

diff --git a/view/base/web/js/price-box.js b/view/base/web/js/price-box.js
index f48ad0f..053d928 100644
--- a/view/base/web/js/price-box.js
+++ b/view/base/web/js/price-box.js
@@ -194,7 +194,7 @@

             this.options.productId = box.data('productId');

-            if (_.isEmpty(prices)) {
+            if (priceHolders.length > 0) {
                 priceHolders.each(function (index, element) {
                     var type = $(element).data('priceType'),
                         amount = parseFloat($(element).data('priceAmount'));
@@ -223,6 +223,13 @@
          * Updates product final and base price according to tier prices
          */
         updateProductTierPrice: function updateProductTierPrice() {
+            var tierPrice = this.getPrice('price'),
+                tierBasePrice = this.getPrice('basePrice');
+
+            if (!tierPrice || !tierBasePrice) {
+                return;
+            }
+
             var originalPrice,
                 prices = {'prices': {}};

@@ -246,6 +253,10 @@
          * @returns {Number}
          */
         getPrice: function (priceKey) {
+            if (!this.options.priceConfig.tierPrices) {
+                return null;
+            }
+
             var productQty = $(this.qtyInfo).val(),
                 result,
                 tierPriceItem,
